### ISSUE TITLE
special character in google drive

### ISSUE THREAD

**Issue by ArthurGing** [rclone/rclone#3083]
#### What is the problem you are having with rclone?
I have two directory, one named '123' under the root, another named '.' under '123'
Then I list it from root with 'lsf' command.
I think it will show '123/' and '123/./', but it shows error

Here is my command:
rclone lsf --format sp -R remote:  --no-check-certificate --exclude="**[\\\\／:\*\?\"<>|]**" --exclude=".*{/**,}"

console:
-1;123/
2019/04/01 18:04:13 ERROR : 123: Entry doesn't belong in directory "123" (too short) - ignoring

#### What is your rclone version (output from `rclone version`)
1.45

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
linux 64 bit

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
lsf

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
2019/04/01 18:11:39 DEBUG : ": Excluded
2019/04/01 18:11:39 DEBUG : >: Excluded
2019/04/01 18:11:39 DEBUG : <: Excluded
2019/04/01 18:11:39 DEBUG : :: Excluded
2019/04/01 18:11:39 DEBUG : ?: Excluded
2019/04/01 18:11:39 DEBUG : .: Excluded
2019/04/01 18:11:39 DEBUG : \: Excluded
2019/04/01 18:11:39 DEBUG : *: Excluded
2019/04/01 18:11:39 DEBUG : |: Excluded
2019/04/01 18:11:39 DEBUG : ／: Excluded
-1;123/
2019/04/01 18:11:39 ERROR : 123: Entry doesn't belong in directory "123" (too short) - ignoring
2019/04/01 18:11:39 DEBUG : 5 go routines active
2019/04/01 18:11:39 DEBUG : rclone: Version "v1.45" finishing with parameters ["rclone" "lsf" "--format" "sp" "-R" "remote:" "--no-check-certificate" "--exclude=**[\\\\／:\\*\\?\"<>|]**" "--exclude=.*{/**,}" "-vv"]

**Comment by ncw** [comment]
I made a directory structure as you suggested.

I can replicate this with v1.45 and v1.46 but not with [the latest beta](https://beta.rclone.org) - that has a different problem.

```
$ rclone lsf --format sp -R drive:test/issue-3083/
-1;123/
-1;123/
```

This change was introduced in d5f2df2f3d1b9708d07b5a29876fde96ab2df392 where we switched from using List to ListR so running this disabling ListR gives the same problem

```
$ rclone lsf --disable ListR --format sp -R drive:test/issue-3083/
-1;123/
2019/04/01 18:14:20 ERROR : 123: Entry doesn't belong in directory "123" (too short) - ignoring
```

So the underlying problem is still present.

The problem seems to be that we use `path.Join` throughout the drive backend which will join "123" and "." to get "123"...

@B4dM4n any thoughts?

**Comment by B4dM4n** [comment]
There are various uses of `path.Join` and `path.Clean` throughout the `rclone` source. As far as I can tell most backends treat paths like Posix paths, inheriting the special meaning of `.` and `..`.

We could stop using `path.Join` and use a custom implementation to avoid the special treatment of `.` and `..` in some backends, but this could yield varying results when using the same path on different backends and result in files not being copyable between backends that treat them differently.

Another solution could be using the filename encoding of #3148 and replace `.` and `..` with `．` and `．．`. I already added this for testing purposes and it worked great in my tests.

This way a folder named `gdrive:123/.` would be listed as `gdrive:123/．` and can be copied with this name to a backend other than Google Drive.

```shell
% mkdir -p 123/．
% date > 123/．/file.txt
% rclone copy 123 TestDrive:123
% rclone tree TestDrive:123
/
└── ．
    └── file.txt

1 directories, 1 files
```

The downside is that writing such a name on the command line requires to use Unicode in the terminal as all special paths in #3148.

**Comment by ncw** [comment]
I've merged the fix for this to master now which means it is in [the latest beta](https://beta.rclone.org) and will be released in v1.50
